internal/controllers/machine/drain/drain.go

```diff
@@ -281,6 +281,7 @@ func (d *Helper) EvictPods(ctx context.Context, podDeleteList *PodDeleteList) Ev
 	var podsToTriggerEvictionLater []PodDelete
 	var podsWithDeletionTimestamp []PodDelete
```

Comment:
Please check the existing test coverage for the modified funcs and extend accordingly for the new case (we should have test coverage for everything, only new cases should be needed)

Reply:
Let's also extend the NodeDrainTimeout e2e test to cover waitcompleted (node_drain.go)

(Unfortunately the whole drain implementation is a bit brittle (already before this PR) so e2e coverage would be really good. For example with the current PR even Pods with the label set to waitcompleted would have been drained because the machineDrainRulesFilter would have overwritten the result of the drainLabelFilter with behavior drain)